components/script/animations.rs

```diff
@@ -138,20 +144,38 @@ impl Animations {
         sets.retain(|_, state| !state.is_empty());
         let have_running_animations = sets.values().any(|state| state.needs_animation_ticks());
 
-        self.update_running_animations_presence(window, have_running_animations);
+        self.update_running_animations_presence(window, Some(have_running_animations));
     }
 
-    fn update_running_animations_presence(&self, window: &Window, new_value: bool) {
-        let have_running_animations = self.have_running_animations.get();
-        if new_value == have_running_animations {
+    fn update_running_animations_presence(
+        &self,
+        window: &Window,
+        new_have_running_animations: Option<bool>,
```

Comment:
Accepting an Option<bool> here makes things a bit confusing, so I would just keep this function more-or-less the same, but add a new helper function:

```rust
    fn update_running_animations_presence(&self, window: &Window, new_value: bool) {
        let had_running_animations = self.has_running_animations.get();
        if new_value == had_running_animations {
            return;
        }

        self.has_running_animations.set(new_value);
        self.handle_animation_presence_or_pending_events_change(window);
    }

    fn handle_animation_presence_or_pending_events_change(&self, window: &Window) {
        let has_running_animations = self.has_running_animations.get();
        let has_pending_events = self.pending_events.borrow().is_empty();

        // Do not send the NoAnimationCallbacksPresent state until all pending
        // animation events are delivered.
        let state = match has_running_animations || has_pending_events {
            true => AnimationsPresentState::AnimationsPresent,
            false => AnimationsPresentState::NoAnimationsPresent,
        };
        window.send_to_constellation(ScriptMsg::ChangeRunningAnimationsState(state));
    }

```